analyzers/src/SonarAnalyzer.CSharp/Rules/MemberInitializerRedundant_RoslynCfg.cs

```diff
@@ -0,0 +1,161 @@
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using SonarAnalyzer.CFG.Helpers;
+using SonarAnalyzer.CFG.Roslyn;
+using SonarAnalyzer.Extensions;
+using StyleCop.Analyzers.Lightup;
+using SymbolWithInitializer = System.Collections.Generic.KeyValuePair<Microsoft.CodeAnalysis.ISymbol, Microsoft.CodeAnalysis.CSharp.Syntax.EqualsValueClauseSyntax>;
+
+namespace SonarAnalyzer.Rules.CSharp
+{
+    public sealed partial class MemberInitializerRedundant
+    {
+        private class MemberInitializerRedundancyCheckerRoslyn : CfgAllPathValidator
+        {
+            private readonly IsValidMatcher isValidMatcher;
+            private readonly IsInvalidMatcher isInvalidMatcher;
+
+            public MemberInitializerRedundancyCheckerRoslyn(ControlFlowGraph cfg, ISymbol memberToCheck, SemanticModel semanticModel) : base(cfg)
+            {
+                var redundancyChecker = new RedundancyChecker(memberToCheck, semanticModel);
+                this.isValidMatcher = new IsValidMatcher(redundancyChecker);
+                this.isInvalidMatcher = new IsInvalidMatcher(redundancyChecker);
+            }
+
+            // Returns true if the block contains assignment before access
+            protected override bool IsValid(BasicBlock block) =>
+                isValidMatcher.Matches(block);
+
+            // Returns true if the block contains access before assignment
+            protected override bool IsInvalid(BasicBlock block) =>
+                isInvalidMatcher.Matches(block);
+
+            private abstract class RedundancyMatcherBase : OperationFinder<bool>
+            {
+                protected readonly RedundancyChecker redundancyChecker;
+
+                protected RedundancyMatcherBase(RedundancyChecker redundancyChecker) =>
+                    this.redundancyChecker = redundancyChecker;
+
+                public bool Matches(BasicBlock block)
+                {
+                    TryFind(block, out var result);
+                    return result;
+                }
+            }
+
+            private  class IsValidMatcher : RedundancyMatcherBase
+            {
+                public IsValidMatcher(RedundancyChecker redundancyChecker) : base(redundancyChecker) { }
+
+                protected override bool TryFindOperation(IOperationWrapperSonar operation, out bool result)
```

Comment:
This whole thing should be way easier and shorter using operations. We should be looking for few operations, like `IAssignmentOperation`. That will directly give us the `Target` symbol and we should not need to work with Syntax here at all. Something like I did in the LVA.

Imagine this part to be common for C# and VB.